pkg/core/coretesting/resources.go

```diff
@@ -0,0 +1,49 @@
+package coretesting
+
+import (
+	"testing"
+
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/graph"
+	"github.com/stretchr/testify/assert"
+)
+
+type (
+	StringDep = graph.Edge[string]
+
+	ResourcesExpectation struct {
+		Nodes []string
+		Deps  []StringDep
+
+		// OnlyContains only check whether the dag contains the `.Nodes` and `.Deps`. If false,
+		// checks full equality.
+		OnlyContains bool
+	}
+)
+
+func (expect ResourcesExpectation) Assert(t *testing.T, dag *core.ResourceGraph) {
+	var res []string
+	for _, r := range dag.ListResources() {
+		res = append(res, r.Id())
+	}
+	if expect.OnlyContains {
+		for _, r := range expect.Nodes {
+			assert.Contains(t, res, r)
+		}
```

Comment:
Would this work?

```suggestion
		assert.Subset(t, res, expect.Nodes)
```

Ditto for the Deps below. 

If so, you might even consider something like:

```
matchFunc := assert.ElementsMatch
if expect.OnlyContains {
   matchFunc = assert.Subset
}
```

...and then you don't need the if-elses in the rest of this function.

Reply:
Ah, thanks - I was looking for this assert function but didn't think to check subset (looking for like a "ContainsAll" or "In").